cmd/server/main.go

```diff
@@ -96,12 +110,13 @@ func main() {
 
 			ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
 			defer cancel()
+
 			if err := s.Server.Shutdown(ctx); err != nil {
 				log.Fatalf("Server shutdown failed: %v", err)
 			}
 			fmt.Println("Server stopped gracefully")
-			err := storage.Backup()
-			if err != nil {
+
+			if err := storage.Backup(); err != nil {
```

Comment:
```suggestion
			if err := storage.BackupPeriodically(); err != nil {
```
По заявленной логике (с учетом предложенных исправлений) `storage.Backup()` будет срабатывать каждый раз, когда включен синхронный бекап (StoreInterval == 0). В этом случае бекап будет консистентным, после шатдауна сервера новых записей в in-memory не появится. В случае, если бекап асинхронный (StoreInterval > 0), в процессе завершения работы он может действительно быть не консистентен состоянию in-memory и действительно необходимо будет сбросить данные на диск. Поэтому нужно дергать именно `BackupPeriodically`, так как эта функция срабатывает именно при условии StoreInterval > 0.

Reply:
готово